sycl/include/CL/sycl/group_algorithm.hpp

```diff
@@ -141,6 +179,28 @@ reduce_over_group(Group, T x, BinaryOperation binary_op) {
 #endif
 }
 
+// complex specialization. T is std::complex<float> or similar.
+//   binary op is  sycl::plus<std::complex<float>>
+template <typename Group, typename T>
+detail::enable_if_t<(is_group_v<std::decay_t<Group>> &&
+                     detail::is_complex<T>::value &&
+                     detail::is_native_op<T, sycl::plus<T>>::value),
```

Comment:
I think we need something like this:
```suggestion
                     (detail::is_native_op<T, sycl::plus<T>>::value ||
                      detail::is_native_op<T, sycl::plus<void>>::value),
```
...because we want to be able to call `reduce_over_group(g, std::complex<float>{}, std::plus<>())`.

Reply:
Good catch. I've updated it so `sycl::plus<>()` works as well and have updated the tests to cover that as well.